pdata/ptrace/jsoniter.go

```diff
@@ -0,0 +1,369 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ptrace // import "go.opentelemetry.io/collector/pdata/ptrace"
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"go.opentelemetry.io/collector/pdata/internal"
+	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+	otlptrace "go.opentelemetry.io/collector/pdata/internal/data/protogen/trace/v1"
+)
+
+// NewJSONIterUnmarshaler returns a model.Unmarshaler. Unmarshals from OTLP json bytes.
+func NewJSONIterUnmarshaler() Unmarshaler {
+	return newJSONIterUnmarshaler()
+}
+
+type jsonIterUnmarshaler struct {
+}
+
+func newJSONIterUnmarshaler() *jsonIterUnmarshaler {
+	return &jsonIterUnmarshaler{}
+}
+
+func (d *jsonIterUnmarshaler) UnmarshalTraces(buf []byte) (Traces, error) {
+	iter := jsoniter.ConfigFastest.BorrowIterator(buf)
+	td := readTraceData(iter)
+	err := iter.Error
+	jsoniter.ConfigFastest.ReturnIterator(iter)
```

Comment:
This should be a "defer" in case we have a check and return faster in the future to not forget about it.

Reply:
Make sense, let me update it.

Reply:
You may have missed this :)

Reply:
Done